app/routes/server-example.tsx

```diff
@@ -0,0 +1,179 @@
+import type { Route } from "./+types/server-example";
+import { prefetch } from "~/lib/prefetch";
+import { ClientDataExample } from "~/components/client-data-example";
+import { appRouter } from "~/api/router";
+import { createContext } from "~/api/context";
+
+export function meta({}: Route.MetaArgs) {
+  return [
+    { title: "React Server Components Example - Jebsite" },
+    {
+      name: "description",
+      content:
+        "Demonstrating React Server Components vs Client Components in React Router v7",
+    },
+  ];
+}
+
+export const unstable_middleware: Route.unstable_MiddlewareFunction[] = [
+  async () => {
+    const { queryClient, trpc } = prefetch();
+
+    // Prefetch some data for client components
+    void queryClient.prefetchQuery(trpc.serverDemo.getStats.queryOptions());
+  },
+];
+
+export async function ServerComponent({}: Route.ComponentProps) {
+  // Create server context for tRPC calls (use a mock request since we don't have access to it here)
+  const mockRequest = new Request("http://localhost/server-example");
+  const ctx = createContext({ req: mockRequest, resHeaders: new Headers() });
+  const caller = appRouter.createCaller(ctx);
```

Comment:
This seems weird, we should use the AsyncContext that provides this, right??

Reply:
Can we just include a trpc caller in the async context and have a helper function for getting that as well? I don't want this boilerplate